clusterversion: set devBranch to false, version to beta.1

celiala opened the pull request:
To be merged directly onto `release-24.3` **(DO NOT MERGE ONTO MASTER)**

As per the runbook to [prepare for 24.3 beta test releases](https://github.com/cockroachdb/cockroach/blob/master/pkg/clusterversion/README.md#r1-prepare-for-beta), this PR:
- sets `DevelopmentBranch=false`
- sets `version.txt=v24.3.0-beta.1`

Epic: REL-1302
Release note: None
Release justification: release-infra process.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132667)
<!-- Reviewable:end -->

celiala committed:
clusterversion: set devBranch to false, version to beta.1

To be merged directly onto `release-24.3` **(DO NOT MERGE ONTO MASTER)**

As per the runbook to [prepare for 24.3 beta test releases](https://github.com/cockroachdb/cockroach/blob/master/pkg/clusterversion/README.md#r1-prepare-for-beta), this PR:
- sets `DevelopmentBranch=false`
- sets `version.txt=v24.3.0-beta.1`

Epic: REL-1320
Release note: None
Release justification: release-infra process.

celiala commented:
blathers backport release-24.3

celiala commented:
https://github.com/cockroachdb/cockroach/pull/132923 has been merged onto `release-24.3`.

we can close this draft now.
